ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixClusterManagerMetrics.java

```diff
@@ -190,15 +196,104 @@ void initializePartitionMetrics() {
+
+      for (PartitionId partition : clusterMapCallback.getPartitions()) {
+        // Get total replica count
+        int totalLocalReplicaCount = (int) partition.getReplicaIds()
+            .stream()
+            .filter(replicaId -> replicaId.getDataNodeId().getDatacenterName().equals(dcName))
+            .count();
+
+        // Get local leader/standby/bootstrap replica counts of this partition
+        int localLeaderStandByReplicaCount = 0;
+        for (List<? extends ReplicaId> replicaIds : partition.getReplicaIdsByStates(
+            EnumSet.of(ReplicaState.LEADER, ReplicaState.STANDBY), dcName).values()) {
+          localLeaderStandByReplicaCount += replicaIds.size();
+        }
+        int localBootstrapReplicaCount = partition.getReplicaIdsByState(ReplicaState.BOOTSTRAP, dcName).size();
+        int localInactiveReplicaCount = partition.getReplicaIdsByState(ReplicaState.INACTIVE, dcName).size();
+
+        // Calculate the number of replicas that are down. Bootstrap (upcoming) and Inactive (deactivating) replicas are considered as good.
+        int downReplicaCount = totalLocalReplicaCount - localLeaderStandByReplicaCount - localBootstrapReplicaCount
+            - localInactiveReplicaCount;
+
+        // Calculate the number of replicas that are not in leader or standby
+        int nonLeaderStandbyReplicaCount = totalLocalReplicaCount - localLeaderStandByReplicaCount;
+
+        // Update the maps to keep track of partition counts where 1, 2 or more replicas are down.
+        if (downReplicaCount > 0) {
+          localPartitionCountsByDownReplicas.put(downReplicaCount,
+              localPartitionCountsByDownReplicas.getOrDefault(downReplicaCount, 0) + 1);
+        }
+        if (nonLeaderStandbyReplicaCount > 0) {
+          localPartitionCountsByNonLeaderStandbyReplicas.put(nonLeaderStandbyReplicaCount,
+              localPartitionCountsByNonLeaderStandbyReplicas.getOrDefault(nonLeaderStandbyReplicaCount, 0) + 1);
+        }
+      }
+
+      return localPartitionCountsByDownReplicas.getOrDefault(1, 0);
+    };
+
+    // Metric to see number of local partitions with 1 replica down
+    registry.gauge(MetricRegistry.name(HelixClusterManager.class, "localPartitionCountWithOneReplicaDown"),
+        () -> getLocalPartitionCountWithOneReplicaDown);
+
+    // Metric to see number of local partitions with 2 replica down
+    Gauge<Integer> getLocalPartitionCountWithTwoReplicaDown =
+        () -> localPartitionCountsByDownReplicas.getOrDefault(2, 0);
+    registry.gauge(MetricRegistry.name(HelixClusterManager.class, "localPartitionCountWithTwoReplicaDown"),
+        () -> getLocalPartitionCountWithTwoReplicaDown);
+
+    // Metric to see number of local partitions with 3 or more replica down
+    Gauge<Integer> getLocalPartitionCountWithThreeOrMoreReplicaDown = () -> {
+      int numOfPartitions = 0;
+      for (int downReplicaCount : localPartitionCountsByDownReplicas.keySet()) {
+        if (downReplicaCount >= 3) {
```

Comment:
Instead of hardcoding the values 3,2,1. Can we have it be a function of the number of replicas in case of local cluster's 4-replica RT and video cluster's cluster 1-replica RT or should we just skip 1-replica RTs since that may skew the metric whenever a host goes down in video

Reply:
Hmm, in 4-replica cluster too, we have count of partitions which lost 1 replica, 2 replica and 3+ replica. May be we could get on the partitions which lost 2 or more replicas? 

Yeah, you are right. For video cluster since we have a mix of 3 and 1 replica partitions, we may ignore 1-replica partitions which lost 1 replica. I am thinking we can fix it later.